Sentinel/Classes/Core/PreferencesInfo/Views/OptionToggleView.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  OptionToggleView.swift
+//  Sentinel
+//
+//  Created by Zvonimir Medak on 27.11.2024..
+//
+
+import SwiftUI
+
+struct OptionToggleView: View {
+
+    @State private var value: Bool
+    @State private var errorMessage: String?
+    let title: String
+    let description: String?
+    let onValueChanged: (Bool) -> Void
+    let getter: () -> Bool
+    let hasErrorMessage: (Bool) -> String?
+
+    var body: some View {
+        VStack(spacing: 8) {
+            Toggle(isOn: $value) {
+                Text(title)
+                    .font(.body1Bold)
+                    .frame(maxWidth: .infinity, alignment: .leading)
+            }
+
+            DescriptionView(description: description, errorMessage: errorMessage)
+        }
+        .onChange(of: value) { value in
+            guard let message = hasErrorMessage(value) else {
+                errorMessage = nil
+                onValueChanged(value)
+                return
+            }
+            errorMessage = message
+        }
+        .onAppear {
+            let fetchedValue = getter()
+            guard fetchedValue != value else { return }
+            value = fetchedValue
```

Comment:
Can this be changed into something like this?
```swift
guard let fetchedValue = getter() else { return }
value = fetchedValue
```

Reply:
I'm not really sure why I added that guard since SwiftUI should handle the updates properly. The guard is not needed here since the getter doesn't return an optional value. So we can reduce it to just `value = getter()`